src/pages.rs

```diff
@@ -35,13 +37,70 @@ impl From<crate::Error> for ErrorResponse<'_> {
 pub struct Index<'a> {
     meta: &'a env::Metadata<'a>,
     base_path: &'static env::BasePath,
+    max_expiry: Option<u32>,
+
+    /// SAFETY: calls in the template are always sequential
+    default_has_been_written: Cell<bool>,
```

Comment:
I'm still thinking if there is a more functional way to express the statefulness of the Askama filter. Feels a bit hackish to be honest. One way would be to generate the HTML in Rust code and just pass it down. Not sure though.

Reply:
Askama templates technically allow for *some* state to be had within the templates, but its very verbose to pass back out to rust functions (see: `Self::expiry(self...)`).

We could just generate the whole expiry set in rust and pass it out that is true. i was first playing with a filter, but that got annoying fast.

I am not too fussed either direction, if you prefer to generate the whole option set in the function itself i can change it.

Reply:
Up to you :-)